parasoft-findings-teamcity-agent/src/main/resources/com/parasoft/findings/teamcity/agent/sa-pmd.xsl

```diff
@@ -0,0 +1,83 @@
+<?xml version="1.0"?>
+<xsl:stylesheet version="1.1" xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns="http://pmd.sourceforge.net/report/2.0.0" >
+    <xsl:output method="xml" encoding="UTF-8" indent="yes" />
+
+    <xsl:variable name="toolName" select="/ResultsSession/@toolName"/>
+    <xsl:variable name="rules" select="/ResultsSession/CodingStandards/Rules/RulesList/Rule"/>
+    <xsl:variable name="categories" select="/ResultsSession/CodingStandards/Rules/CategoriesList//Category"/>
+    <xsl:variable name="isStaticAnalysisResult" select="count(/ResultsSession/CodingStandards) = 1" />
+
+    <xsl:template match="/">
+        <xsl:choose>
+            <xsl:when test="$isStaticAnalysisResult">
+                <xsl:element name="pmd">
+                    <xsl:call-template name="transformStdViols"/>
+                    <xsl:call-template name="transformSuppressedViols"/>
+                </xsl:element>
+            </xsl:when>
+        </xsl:choose>
+    </xsl:template>
+
+    <xsl:template name="transformStdViols">
+        <xsl:for-each select="/ResultsSession/CodingStandards/StdViols/StdViol">
+            <xsl:choose>
+                <xsl:when test="@supp != true()">
+                    <xsl:element name="file">
+                        <xsl:attribute name="name">
+                            <xsl:value-of select="@locFile"/>
+                        </xsl:attribute>
+                        <xsl:call-template name="StdViol"/>
+                    </xsl:element>
+                </xsl:when>
+            </xsl:choose>
+        </xsl:for-each>
+    </xsl:template>
+
+    <xsl:template name="StdViol">
+        <xsl:element name="violation">
+            <xsl:attribute name="beginline"><xsl:value-of select="@locStartln" /></xsl:attribute>
+            <xsl:attribute name="endline"><xsl:value-of select="@locEndLn"/></xsl:attribute>
+            <xsl:attribute name="begincolumn"><xsl:value-of select="@locStartPos"/></xsl:attribute>
+            <xsl:attribute name="endcolumn"><xsl:value-of select="@locEndPos"/></xsl:attribute>
+            <xsl:attribute name="priority"><xsl:value-of select="@sev"/></xsl:attribute>
+            <xsl:attribute name="rule">
+                <xsl:value-of select="@rule"/>
+            </xsl:attribute>
+            <xsl:attribute name="ruleset">
+                <xsl:call-template name="getRuleCategoryDesc">
+                    <xsl:with-param name="ruleId" select="@rule"/>
+                </xsl:call-template>
+            </xsl:attribute>
+            <xsl:attribute name="package"><xsl:value-of select="@pkg"/></xsl:attribute>
```

Comment:
"pkg" here is an optional value. We should only add it when it exists in Parasoft report. 
Consider to use `<xsl:if test="@pkg">` for case like this.

Please check the schema file here: [reports.xsd](https://jira.parasoft.com/secure/attachment/105586/reports.xsd)
e.g. For "StdViol", search for `<xs:attributeGroup name="attlist.StdViol">` and make sure we do the existence check for all optional attributes. (default is "optional" or look at attribute "use" if exists)

Reply:
Please also do the check for sa-pmdcpd.xsl